### ISSUE TITLE
Both rivals can "Win The Rivalry Game" if they play twice in a year

### ISSUE THREAD

**Issue by destilla** [jonesguy14/footballcoach#59]
TL;DR at bottom

Had this happen earlier in hard mode (and I'm pretty sure the introduction of hard mode is the only reason I caught it) --

Basically what happens is you play your rival, someone wins. In playGame(), after the game is over, a check is made to see if the teams are rivals. If they are, the winner has wonRivalryGame set to true.

If they play a second or third time (CCG, Bowl Game) and the team that has yet to win is the winner in one of the rematches, their wonRivalryGame is set to true. Both teams get prestige and if it's a user team they're shown the "Hooray you beat your rival" message.

I didn't screenshot this because I was on my phone in a meeting when it happened, and duplicating it has been a pain because it requires my rival to be good enough to make it to the CCG even after losing to me (or for me to be good enough to make it after losing to them). 

In my scenario, I lost the regular season matchup, met them and won in the CCG and went on to win the NCG. I checked the post season message and it congratulated me for beating my rival.

----TL;DR----

Thought about it, checked code, saw we don't currently handle multiple rivalry match-ups in a single season.

**Comment by jonesguy14** [comment]
Yeah that's actually on purpose! Originally I had it so only the regular season match-up mattered, but a couple people on the subreddit didn't like it when they would beat their rival in the CCG or even the NCG and didn't get the rivalry bonus prestige. So I changed it so any match up between rivals will award prestige.

**Comment by destilla** [comment]
Ahhh that does make sense.

My new question then becomes: Should this still be the case on hard mode?

And also, if I choose hard mode and I beat my rival, but they end up being ranked like #50, should I still get the prestige? I know right it's just prestige check (if your rival is within 20 prestige you get it, otherwise no bonus), so hard mode is an automatic bonus if you win (because the rival is always within 10).

There have been several season where I rofl stomp my rival and they end up being ranked super low while I go on to win the NCG or something close to it. And then I get a bonus for beating my rival, who very clearly were not a good team that year.

**Comment by akeaswaran** [comment]
I saw this issue in the iOS version as well. Could we could implement some sort of exponential addition to the rivalry prestige bonus depending on how many times you beat/lose to your rival in a given season? Something like:

* 1 win in a 1-rivalry game season: +2
* 2 wins in 2-rivalry game season: +3
* 3 wins in 3-rivalry game season: +4
(and possibly something else for 2 wins in a 3-game season)

OR we could check the winner of the season series and just award the two prestige points that way. That might be a way to shortcut all of the additional logic the first solution may require.

**Comment by jonesguy14** [comment]
Yeah maybe I should change it for Hard Mode, I've noticed that too where a team was pretty bad but their prestige was still decent so a bonus was awarded.

A tiered system would be cool too, so if you beat them more times, you get more prestige (and vice versa for losing).
